peerconnection.go

```diff
@@ -326,14 +326,68 @@ Once the connection succeeds, .OnDataChannel should trigger on the remote peer's
 |PeerConnection|, while .OnOpen should trigger on the local DataChannel returned
 by this method. Both DataChannel references should then be open and ready to
 exchange data.
-
-TODO: Implement the "negotiated" flag?
 */
-func (pc *PeerConnection) CreateDataChannel(label string, dict Init) (
+
+// Ordered configures a DataChannels 'ordered' option.
+func Ordered(ordered bool) func(*DataChannelInit) {
+	return func(i *DataChannelInit) {
+		i.Ordered = ordered
+	}
+}
+
+// MaxRetransmitTime configures a DataChannels 'maxRetransmitTime' option.
+func MaxRetransmitTime(maxRetransmitTime int) func(*DataChannelInit) {
+	return func(i *DataChannelInit) {
+		i.MaxRetransmitTime = maxRetransmitTime
+	}
+}
+
+// MaxRetransmits configures a DataChannels 'maxRetransmits' option.
+func MaxRetransmits(maxRetransmits int) func(*DataChannelInit) {
+	return func(i *DataChannelInit) {
+		i.MaxRetransmits = maxRetransmits
+	}
+}
+
+// Negotiated configures a DataChannels 'negotiated' option.
+func Negotiated(negotiated bool) func(*DataChannelInit) {
+	return func(i *DataChannelInit) {
+		i.Negotiated = negotiated
+	}
+}
+
+func (pc *PeerConnection) CreateDataChannel(label string, options ...func(*DataChannelInit)) (
```

Comment:
I'm not sure how I feel about making this varidic.  Is this a common pattern for handling default options?

What if we didn't expose `DataChannelInit` but instead exported a constructor for it that filled in those values?

```
func NewDataChannelInit() {
  return dataChannelInit{
    Ordered:           true,
    MaxRetransmitTime: -1,
    MaxRetransmits:    -1,
    Negotiated:        false,
    ID:                -1,
  }
}

init := NewDataChannelInit()
init.Ordered = false
dc := CreateDataChannel("label", init)
```

Reply:
> I'm not sure how I feel about making this varidic. Is this a common pattern for handling default options?

It seems like `ConfigurationOption` serves a similar purpose, so there's a precedent for this already in the project.

Reply:
And there's also this breadcrumb ...

> // TODO: Variadic options constructor, probably makes more sense for
> // CreateDataChannel in parent package PeerConnection.